columnflow/production/cms/pdf.py

```diff
@@ -74,4 +74,26 @@ def pdf_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
     events = set_ak_column_f32(events, "pdf_weight_up", 1 + stddev)
     events = set_ak_column_f32(events, "pdf_weight_down", 1 - stddev)
 
+    if ak.any(stddev > 0.5):
+        # catch events with large pdf variations
+        # TODO: how should we handle large PDF weight uncertainties (especially above 100%)?
+        occurances = ak.sum(stddev > 0.5)
+        frac = occurances / ak.count(stddev, axis=0) * 100
+        logger.warning(
+            f"In dataset {self.dataset_inst.name}, there are {occurances} ({frac:.2f}%) " +
+            "entries with pdf uncertainty above 50%",
+        )
+
+    if ak.any(pdf_weight_nominal == 0):
```

Comment:
Same as above, (e.g.) `zero_mask = pdf_weight_nominal == 0`.

From the PR title I understand that nan's are an issue at some point. Where do they appear? I'm wondering if setting everything (nominal + variations) to zero helps.

Reply:
They appear when the 0th pdf weight entry (pdf_weight_nominal) is 0, on which we normalize here: https://github.com/columnflow/columnflow/blob/17d2d12b699cd94d0e554692f3b3fc3e803b7d79/columnflow/production/cms/pdf.py#L66